frame/wormhole/backing/s2s/src/lib.rs

```diff
@@ -224,32 +247,31 @@ pub mod pallet {
 		) -> DispatchResultWithPostInfo {
 			let user = ensure_signed(origin)?;
 
-			if let Some(fee_account) = T::FeeAccount::get() {
-				T::RingCurrency::transfer(&user, &fee_account, fee, KeepAlive)?;
-			}
-			let token = Token::Native(TokenInfo {
+			let token_metadata = TokenMetadata {
+				token_type: NATIVE_TOKEN_TYPE,
 				address: T::RingPalletId::get().into_h160(),
-				value: None,
-				option: Some(TokenOption {
-					name: RING_NAME.to_vec(),
-					symbol: RING_SYMBOL.to_vec(),
-					decimal: RING_DECIMAL,
-				}),
-			});
-			let payload =
-				T::CallEncoder::encode_remote_register(spec_version, weight, token.clone());
-			T::MessageSender::send_message_by_root(
-				T::MessageSendPalletIndex::get(),
+				name: RING_NAME.to_vec(),
+				symbol: RING_SYMBOL.to_vec(),
+				decimal: RING_DECIMAL,
+			};
```

Comment:
Give a `TokenMetadata::new` function?

Reply:
All the member is public, and I find it cannot become more simple to add a initialize function for this struct. What about your opinion?

Reply:
> I find it cannot become more simple to add a initialize function for this struct
The reason is ?

Reply:
every member needs to be a parameter

Reply:
Give it a new to avoid writing field1..2

```rs
MyStruct {
    field1: xxx
    field2: xxx
}
```